session-helper/background-monitor.c

```diff
@@ -28,12 +28,86 @@
 #include <libnotify/notify.h>
 #include "background-monitor.h"
 #include "app/flatpak-permission-dbus-generated.h"
+#include "app/gnome-shell-introspect-dbus-generated.h"
 
 #define PERMISSION_TABLE "background"
 #define PERMISSION_ID "background"
 
 typedef enum { UNSET, NO, YES, ASK } Permission;
 
+static GnomeShellIntrospect *shell = NULL;
```

Comment:
There's an obvious question here: do we want to depend on a gnome-shell api directly, or funnel this through portal backends / frontends ?

Reply:
Yeah, hardcoding a gnome specific api here feels a bit wrong. Several choices here: indirection via portal, indirection via abstraction in the flatpak code, and specifying a cross-desktop API for this.

Reply:
Also, the api used is not an ideal match for the usecase. I can easily imagine some desktop having an api for this that is more targeted than just "show all open windows", so we should be careful about encoding in the apis/abstractions to expose to much details of the current gnome-shell api.